src/raftstore/store/peer.rs

```diff
@@ -1211,6 +1211,12 @@ impl Peer {
 
         self.add_ready_metric(&ready, &mut ctx.raft_metrics.ready);
 
+        if !ready.committed_entries.as_ref().map_or(true, Vec::is_empty)
+            && ctx.current_time.is_none()
+        {
+            ctx.current_time.replace(monotonic_raw_now());
```

Comment:
why not do it when observing commit log duration

Reply:
if we call monotonic_raw_now() when observing commit log duration, the observed duration may become longer, because between ready_since() and monotonic_raw_now() we need to append raft log to disk which may consume long time